src/framework/preferences/js/PrefsEditor.js

```diff
@@ -362,38 +363,60 @@ var fluid_2_0 = fluid_2_0 || {};
         // and this implementation doesn't seem to be causing a problem at present so we had
         // just better leave it the way it is for now.
         that.applier.change("", completeModel);
+        if (eventName) {
+            that.events[eventName].fire(that);
+        }
         that.events.onPrefsEditorRefresh.fire();
         that.applyChanges();
     };
 
     /**
      * Saves the current model and fires onSave
-     */
+ */
     fluid.prefs.prefsEditor.save = function (that) {
-        var savedSelections = fluid.copy(that.model);
+        if (!that.model) {  // Don't save a reset model
+            return;
+        }
 
-        fluid.each(savedSelections, function (value, key) {
-            if (fluid.get(that.initialModel, key) === value) {
-                delete savedSelections[key];
-            }
-        });
-        that.events.onSave.fire(savedSelections);
-        that.setSettings(savedSelections);
+        var initialModel = that.initialModel,
+            userSelections = fluid.copy(that.model);
+
+        // Only save the changed preferences so the future default value change on preferences can still be correctly merged with changed preferences
+        if (fluid.model.diff(userSelections.preferences, initialModel.preferences)) {
+            delete userSelections.preferences;
+        } else {
+            fluid.each(userSelections.preferences, function (value, path) {
+                if (fluid.model.diff(value, fluid.get(that.initialModel, ["preferences", path]))) {
+                    delete userSelections.preferences[path];
+                }
+            });
+        }
+
+        that.events.onSave.fire(userSelections);
+        that.setSettings(userSelections);
+        return userSelections;
     };
 
     fluid.prefs.prefsEditor.saveAndApply = function (that) {
-        that.save();
-        that.events.onPrefsEditorRefresh.fire();
-        that.applyChanges();
+        var prevSettings = that.getSettings(),
+            newSelections = that.save();
+
+        // Only when preferences are changed, re-render panels and trigger enactors to apply changes
+        if (!fluid.model.diff(fluid.get(newSelections, "preferences"), fluid.get(prevSettings, "preferences"))) {
```

Comment:
There's no need for an expensive diff here - the plan was always just to have a standard model listener listening at path "preferences"